erts/emulator/beam/erl_init.c

```diff
@@ -162,7 +162,7 @@ int erts_initialized = 0;
 
 int H_MIN_SIZE;			/* The minimum heap grain */
 int BIN_VH_MIN_SIZE;		/* The minimum binary virtual*/
-int H_MAX_SIZE;			/* The maximum heap size */
+Uint H_MAX_SIZE;		/* The maximum heap size */
```

Comment:
```suggestion
Sint H_MAX_SIZE;		/* The maximum heap size */
```

Reply:
This makes H_MAX_SIZE signed, and max_heap_size is `Uint` everywhere else. I'll keep the Uint, and use Sint only for initial conversion.